docs/FunctionalTestBuildScripts.md

```diff
@@ -70,6 +70,36 @@ A simple hello world program, compiled with `clang 14.0.0` that generates a .exe
 `clang-cl -o clangcl.14.pe.c.codeview.pdbpagesize_default.exe -fuse-ld=lld-link helloc.c -m32 -Z7 -MTd /link /CETCOMPAT /guard:cf /PDB:clangcl.14.pe.c.codeview.pdbpagesize_default.exe.pdb`  
 `clang-cl -o clangcl.14.pe.c.codeview.pdbpagesize_8192_pdbmissing.exe -fuse-ld=lld-link helloc.c -m32 -Z7 -MTd /link /CETCOMPAT /guard:cf /PdbPageSize:8192 /PDB:clangcl.14.pe.c.codeview.pdbpagesize_8192_pdbmissing.exe.pdb (then delete the pdb)`
 
+## gcc.example2.fnostackprotector
+
+A basic C program. Any simple example could work, but the exact source used was:
+
+```c
+#include <stdio.h>
+
+char *foo(int arg1, int arg2, FILE *f)
+{
+	char x[0x1000] = {0};
+	return fgets(x, 0x1000, f);
+}
+
+int main(int argc, char **argv, char **envp)
+{
+	foo(argc, argv, envp);
+	return 0;
+}
+```
+
+Compiled with gcc 9.4.0 using:
+`gcc -g -fno-stack-protector example2.c -o gcc.example2.fnostackprotector`
+
+## gcc.example2.fstackprotectorall+fnostackprotector
+
+This uses the same source as gcc.example2.fnostackprotector.
+
+Compiled with gcc 9.4.0 using:
+`gcc -g -fstack-protector-all -fno-stack-protector binskim_test.c -o gcc.example2.fstackprotectorall+fnostackprotector`
+
```

Comment:
in you latest checkin, 
1. I see the old file is deleted, can you see if you can keep it?
2. I don't see a new file added, I think we should delete this info in the md file , if you add file please use new name. For old test files not added by you no need to add info in md file.
3. fyi `-fstack-protector-all -fno-stack-protector` together was not the way, it was 2 file one with `-fstack-protector-all` and the other with `-fno-stack-protector`.

Reply:
@connorcashmsft please wait, let me work on the tests for you.

Reply:
I will create a PR to your branch. Hold on.

Reply:
since you added two new code path :
no dwarf -> pass
no dwarf -> fail
, the test file you added is still to the old path that has dwarf , I am working on that as well. I will create a PR to your branch today.

Reply:
Created a PR to your branch with my thoughts.
also I fixed a from 2018 old bug that is related to our tests.  :)
[https://github.com/connorcashmsft/binskim/pull/1](https://github.com/connorcashmsft/binskim/pull/1)

Reply:
I just merged in your changes.